### PR TITLE
python@3.9 3.9.6

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

File: Formula/python@3.9.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class PythonAT39 < Formula
   desc "Interpreted, interactive, object-oriented programming language"
   homepage "https://www.python.org/"
   # Keep in sync with python-tk@3.9.
```
Comment (bayandin) [inline Formula/python@3.9.rb:4]: Please update `python-tk@3.9` formula as well (in a different commit in the same PR)

**Review by bayandin — COMMENTED** [review]
There're new version of resources available: `setuptools` 57.0.0 and `pip` 21.1.3

**Comment by carlocab** [comment]
Do you mind cherry-picking my commit from https://github.com/Homebrew/homebrew-core/pull/80052 so we don't have to re-run CI for it?

**Comment by branchvincent** [comment]
> Do you mind cherry-picking my commit from #80052 so we don't have to re-run CI for it?

certainly, hope I did that right 😅

**Comment by carlocab** [comment]
Excellent, thank you!

**Comment by carlocab** [comment]
ARM

```
brew install --only-dependencies cake
brew install cake
brew test --retry --verbose easyengine
brew test --retry --verbose lean
brew test --retry --verbose mathlibtools
brew test --retry --verbose monkeysphere
brew install --only-dependencies paket
brew install paket
brew test --retry --verbose spack
```

Big Sur

```
Error: 15 failed steps!
brew test --retry --verbose asuka
brew test --retry --verbose corral
brew test --retry --verbose creduce
brew test --retry --verbose easyengine
brew test --retry --verbose hyperkit
brew test --retry --verbose influxdb
brew test --retry --verbose lean
brew test --retry --verbose llvm@7
brew test --retry --verbose llvm@8
brew test --retry --verbose llvm@9
brew test --retry --verbose mathlibtools
brew test --retry --verbose monkeysphere
brew test --retry --verbose ponyc
brew test --retry --verbose spack
brew test --retry --verbose wasm-pack
```

Catalina

```
Error: 20 failed steps!
brew install --only-dependencies creduce
brew install creduce
brew test --retry --verbose easyengine
brew install --only-dependencies --include-test haskell-language-server
brew test --retry --verbose haskell-language-server
brew test --retry --verbose hyperkit
brew install --only-dependencies idris
brew install idris
brew test --retry --verbose lean
brew install llvm@7
brew install llvm@8
brew install llvm@9
brew test --retry --verbose mathlibtools
brew install --only-dependencies oclgrind
brew install oclgrind
brew install --only-dependencies odin
brew install odin
brew test --retry --verbose spack
brew test --retry --verbose subversion
brew test --retry --verbose wasm-pack
```

Mojave

```
Error: 8 failed steps!
brew test --retry --verbose easyengine
brew test --retry --verbose hyperkit
brew test --retry --verbose lean
brew test --retry --verbose mathlibtools
brew test --retry --verbose opencascade
brew test --retry --verbose spack
brew test --retry --verbose swift
brew test --retry --verbose wasm-pack
```

Most of these aren't unexpected, but I think we need to take a look at `spack`, at least. Plus all the extra Catalina failures, but that might just be an `out of space on device` error.

**Comment by branchvincent** [comment]
`spack` is a legimate issue, filed https://github.com/spack/spack/issues/24644

**Comment by carlocab** [comment]
Looks like the spack issue will be fixed by spack/spack#24686. In the meantime, maybe we can just make spack depend on `python@3.8`.

**Comment by branchvincent** [comment]
Agreed, I'm not near my computer but I did try the latest `python@3.8` locally and `spack` was indeed happy

**Review by carlocab — APPROVED** [review]
The remaining failures look pre-existing (cf. #79624). Thanks for your work here, @branchvincent!

**Pull request closed without merging**